src/iosrv/srv.c

```diff
@@ -482,12 +598,16 @@ dss_srv_handler(void *arg)
 		if (dx->dx_main_xs)
 			bio_nvme_poll(dmi->dmi_nvme_ctxt);
 
-		if (dss_xstream_exiting(dx))
+		if (dss_xstream_exiting(dx)) {
+			dss_check_timeout_list();
```

Comment:
why do this check in progress ult? I think prior version is better (do the check in sched code)

Reply:
ah, the reason is that check_timeout_list() needs tls being initialized, which is in progress ULT. I guess we can add some "barrier" in check_timeout_list().